tools/ping/src/cli.rs

```diff
@@ -0,0 +1,229 @@
+    /// is not one of "mainnet", "testnet" or "shardnet"
+    genesis_hash: Option<String>,
+    #[clap(long)]
+    /// head height to use in the Handshake we send. This must be provided if --chain-id
+    /// is not one of "mainnet", "testnet" or "shardnet"
+    head_height: Option<u64>,
+    /// Protocol version to advertise in our handshake
+    #[clap(long)]
+    protocol_version: Option<u32>,
+    /// node public key and socket address in the format {pub key}@{socket addr}. e.g.:
+    /// ed25519:7PGseFbWxvYVgZ89K1uTJKYoKetWs7BJtbyXDzfbAcqX@127.0.0.1:24567
+    #[clap(long)]
+    peer: String,
+    /// ttl to set on our Routed messages
+    #[clap(long, default_value = "100")]
+    ttl: u8,
+    /// milliseconds to wait between sending pings
+    #[clap(long, default_value = "1000")]
+    ping_frequency_millis: u64,
+    /// line-separated list of accounts to filter on.
+    /// We will only try to send pings to these accounts
+    // TODO: add --track-validators or something, that will constantly keep track of
+    // which accounts are validators and ping those
+    #[clap(long)]
+    account_filter_file: Option<PathBuf>,
+    /// filename to append CSV data to
+    #[clap(long)]
+    latencies_csv_file: Option<PathBuf>,
+}
+
+fn display_stats(stats: &mut [(crate::PeerIdentifier, crate::PingStats)], peer_id: &PeerId) {
+    let mut acc_width = "account".len();
+    for (peer, _) in stats.iter() {
+        acc_width = std::cmp::max(acc_width, format!("{}", peer).len());
+    }
+    // the ones that never responded should end up at the top with this sorting, which is a
+    // little weird, but we can fix it later.
+    stats.sort_by(|(_, left), (_, right)| left.average_latency.cmp(&right.average_latency));
+    println!(
+        "{:<acc_width$} | {:<10} | {:<10} | {:<17} | {:<17} | {:<17}",
+        "account",
+        "num pings",
+        "num pongs",
+        "min ping latency",
+        "max ping latency",
+        "avg ping latency"
+    );
+    for (peer, stats) in stats.iter() {
+        if stats.pings_sent == 0 {
+            continue;
+        }
+        println!(
+            "{:<acc_width$} | {:<10} | {:<10} | {:<17?} | {:<17?} | {:<17?}{}",
+            peer,
+            stats.pings_sent,
+            stats.pongs_received,
+            stats.min_latency,
+            stats.max_latency,
+            stats.average_latency,
+            if peer_id == &peer.peer_id { " <-------------- direct pings" } else { "" }
+        );
+    }
+}
+
+struct ChainInfo {
+    chain_id: &'static str,
+    genesis_hash: CryptoHash,
+    head_height: BlockHeight,
+}
+
+static CHAIN_INFO: &[ChainInfo] = &[
+    ChainInfo {
+        chain_id: "mainnet",
+        genesis_hash: CryptoHash([
+            198, 253, 249, 28, 142, 130, 248, 249, 23, 204, 25, 117, 233, 222, 28, 100, 190, 17,
+            137, 158, 50, 29, 253, 245, 254, 188, 251, 183, 49, 63, 20, 134,
+        ]),
+        head_height: 71112469,
```

Comment:
how is this value better than, for example, 0?

Reply:
true, no good reason